File: server/models/Order.js
const mongoose = require('mongoose');

const itemSchema = new mongoose.Schema({
  id: { type: String, required: true },
  name: { type: String, required: true },
  price: { type: Number, required: true },
  quantity: { type: Number, required: true },
  image: { type: String, required: true },
});

const orderSchema = new mongoose.Schema({
  userId: {
    type: String,
    required: true,
  },
  items: {
    type: [itemSchema],
    required: true,
  },
  totalPrice: {
    type: Number,
    required: true,
  },
  shippingName: {
    type: String,
    required: true,
  },
  shippingAddress: {
    type: String,
    required: true,
  },
  city: {
    type: String,
    required: true,
  },
  postalCode: {
    type: String,
    required: true,
  },
  shippingPhone: {
    type: String,
    required: true,
  },
  createdAt: {
    type: Date,
    default: Date.now,
  },
});

module.exports = mongoose.model('Order', orderSchema);

File: client/lib/models/order.dart
import 'dart:convert';
import 'package:client/models/cart_item.dart';

class Order {
  final String id;
  final List<CartItem> items;
  final double totalPrice;
  final String shippingName;
  final String shippingAddress;
  final String city;
  final String postalCode;
  final String shippingPhone;
  final String userId;
  final DateTime createdAt;

  Order({
    required this.id,
    required this.items,
    required this.totalPrice,
    required this.shippingName,
    required this.shippingAddress,
    required this.city,
    required this.postalCode,
    required this.shippingPhone,
    required this.userId,
    required this.createdAt,
  });

  Map<String, dynamic> toMap() {
    return {
      'id': id,
      'items': jsonEncode(items.map((item) => item.toMap()).toList()),
      'totalPrice': totalPrice,
      'shippingName': shippingName,
      'shippingAddress': shippingAddress,
      'city': city,
      'postalCode': postalCode,
      'shippingPhone': shippingPhone,
      'userId': userId,
      'createdAt': createdAt.toIso8601String(),
    };
  }

  factory Order.fromMap(Map<String, dynamic> map) {
    return Order(
      id: map['id'],
      items:
          (jsonDecode(map['items']) as List)
              .map((item) => CartItem.fromMap(item))
              .toList(),
      totalPrice: map['totalPrice'],
      shippingName: map['shippingName'],
      shippingAddress: map['shippingAddress'],
      city: map['city'],
      postalCode: map['postalCode'],
      shippingPhone: map['shippingPhone'],
      userId: map['userId'],
      createdAt: DateTime.parse(map['createdAt']),
    );
  }
}
